### ISSUE TITLE
Buttons inside semantic headings

### ISSUE THREAD

**Issue by steverep** [home-assistant/frontend#14143]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

The following locations have buttons incorrectly placed inside an HTML heading tag, which makes screen readers extra chatty and can confuse some assistive technology:

- Any dialog that uses the `createCloseHeading` function has the close button inside the heading
- Device detail page, buttons to create automation, scene, and script are all inside the heading

There might be more as it's not easy to do a global search for.

### Describe the behavior you expected

    The buttons should be outside the heading tags.

### Steps to reproduce the issue

n/a

### What version of Home Assistant Core has the issue?

2022.10.x

### In which browser are you experiencing the issue with?

n/a

### Which operating system are you using to run this browser?

n/a

**Comment by steverep** [comment]
Still an issue

**Comment by steverep** [comment]
Still an issue
